### ISSUE TITLE
Annotation popover not opening when portable text is in an array of objects

### ISSUE THREAD

**Issue by paullacour** [sanity-io/sanity#4333]
**Describe the bug**
When trying to create an annotation to portable text in an object that is inside an array, the annotation popover doesn't open.
I tried editing the options for my annotation to have dialog instead of popover, but it doesn't work either. When using the exact same field config at the document root level, the annotation popover opens just fine.

**To Reproduce**
1. Create a document with a field that is an array of objects
2. This object should have a field with portable text with annotation (either custom annotations or default ones).
3. Create an object, type some content in the portable text editor and try annotating (link) a word or sentence.
4. The popover modal for editing the link should open, but it doesn't.
5. It occurs with this custom portable text setup as well as the default one with default link annotation.

```
    defineField({
      name: 'contactBlocks',
      title: 'Contact Blocks',
      type: 'array',
      of: [
        {
          name: 'contactBlock',
          title: 'Contact Block',
          type: 'object',
          fields: [
            defineField({
              name: 'title',
              title: 'Title',
              type: 'string',
              validation: (Rule) => Rule.required(),
            }),
            defineField({
              name: 'content',
              title: 'Content',
              type: 'body',
            }),
          ],
        },
      ],
      group: 'content',
    }),
```
```
import {defineType} from 'sanity'
import React from 'react'
import annotationLinkEmail from '../annotations/annotationLinkEmail'
import annotationLinkExternal from '../annotations/annotationLinkExternal'

const uppercaseIcon = () => <span style={{fontWeight: 'bold'}}>AA</span>
const uppercaseRender = (props) => (
  <span style={{textTransform: 'uppercase'}}>{props.children}</span>
)

export default defineType({
  name: 'body',
  title: 'Body',
  type: 'array',
  of: [
    {
      type: 'block',
      styles: [
        // {title: 'Normal', value: 'normal'}
      ],
      lists: [
        {title: 'Bullet', value: 'bullet'},
      ],
      marks: {
        annotations: [annotationLinkEmail, annotationLinkExternal],
        decorators: [
          {
            title: 'Strong',
            value: 'strong',
          },
          {
            title: 'Italic',
            value: 'em',
          },
          {
            title: 'Uppercase',
            value: 'uppercase',
            icon: uppercaseIcon,
            component: uppercaseRender,
          },
        ],
      },
    },
  ],
})
```

**Expected behavior**
The annotation popover should open same way it does when the field is at the document level.

**Screenshots**
<img width="688" alt="Screenshot 2023-03-30 at 10 37 40" src="https://user-images.githubusercontent.com/46460166/228792232-2af27777-844a-4fb9-9058-e6604b7b4268.png">

**Which versions of Sanity are you using?**
`3.8.1`

**What operating system are you using?**
`macOS Ventura 13.1`

**Which versions of Node.js / npm are you running?**
`8.19.2`
`v16.18.1`

**Additional context**
`Chrome Version 111.0.5563.146`

**Comment by ndimatteo** [comment]
Also experiencing this on `3.8.x`, had to roll back to `3.7.x` to regain the ability to edit/add annotations.

**Comment by ehowey** [comment]
Just a note to say I am seeing some odd behaviour here too with annotation popovers. The popover is opening for me, but is partially hidden sometimes.

**Comment by robinpyon** [comment]
Thanks for the detailed report, @paullacour!

This issue should be fixed in [3.8.2](https://github.com/sanity-io/sanity/releases/tag/v3.8.2) – I'm unable to repro this with the schema you've supplied (on 3.8.2+) – please give that a go and do shout if you're still having problems here.

@ehowey Please create a separate issue with a reproduction case and we'd be more than happy to look into this for you!

**Comment by paullacour** [comment]
@robinpyon Yes, it is now fixed in `3.8.2` ! Thanks 💯

**Comment by ehowey** [comment]
Ditto - appears to be working for me too.
